File: lab/cppproptest/src/poc_basic.cpp
#include "proptest/proptest.hpp"

#include <cstdlib>
#include <iostream>

int main()
{
    using namespace proptest;

    auto last = forAll(
        [](int x, int y) {
            PROP_ASSERT(x + y == y + x);
            return true;
        },
        {
            .seed = 42,
            .numRuns = 64,
        });

    if (!static_cast<bool>(last)) {
        std::cerr << "poc_basic: property failed\n";
        return EXIT_FAILURE;
    }

    std::cout << "poc_basic: property passed\n";
    return EXIT_SUCCESS;
}

File: lab/python-proptest/src/poc_basic.py
"""Minimal lab script: depends on python-proptest (see ../requirements.txt)."""

from __future__ import annotations

from python_proptest import Gen, run_for_all


def main() -> None:
    ok = run_for_all(
        lambda x, y: x + y == y + x,
        Gen.int(),
        Gen.int(),
        num_runs=64,
        seed=42,
    )
    if not ok:
        raise SystemExit("poc_basic: property failed")
    print("poc_basic: property passed")


if __name__ == "__main__":
    main()
